src/components/Button/index.tsx

```diff
@@ -16,195 +15,155 @@ import themeColors from '@styles/themes/default';
 import CONST from '@src/CONST';
 import validateSubmitShortcut from './validateSubmitShortcut';
 
-const propTypes = {
+type ButtonProps = {
     /** Should the press event bubble across multiple instances when Enter key triggers it. */
-    allowBubble: PropTypes.bool,
+    allowBubble?: boolean;
 
     /** The text for the button label */
-    text: PropTypes.string,
+    text?: string;
 
     /** Boolean whether to display the right icon */
-    shouldShowRightIcon: PropTypes.bool,
+    shouldShowRightIcon?: boolean;
 
     /** The icon asset to display to the left of the text */
-    icon: PropTypes.func,
+    icon?: React.FC<SvgProps> | null;
 
     /** The icon asset to display to the right of the text */
-    iconRight: PropTypes.func,
+    iconRight?: React.FC<SvgProps>;
 
     /** The fill color to pass into the icon. */
-    iconFill: PropTypes.string,
+    iconFill?: string;
 
     /** Any additional styles to pass to the left icon container. */
-    // eslint-disable-next-line react/forbid-prop-types
-    iconStyles: PropTypes.arrayOf(PropTypes.object),
+    iconStyles?: Array<StyleProp<ViewStyle>>;
```

Comment:
Just `StyleProp<ViewStyle>` should be enough, then in the code replace `...iconStyles` with just `iconStyles`, it should work the same way as before
```suggestion
    iconStyles?: StyleProp<ViewStyle>;
```